cvmfs/libcvmfs_int.cc

```diff
@@ -374,6 +374,107 @@ int LibContext::ListDirectory(
   return 0;
 }
 
+int LibContext::GetNestedCatalogAttr(
+  const char *c_path,
+  struct cvmfs_nc_stat *info
+) {
+  ClientCtxGuard ctxg(geteuid(), getegid(), getpid());
+  LogCvmfs(kLogCvmfs, kLogDebug,
+    "cvmfs_stat_nested_catalog (cvmfs_nc_stat) : %s", c_path);
+
+  PathString p;
+  p.Assign(c_path, strlen(c_path));
+
+  /* Find the nested catalog from the root catalog */
+  shash::Any hash;
+  uint64_t size;
+  const bool found =
+    mount_point_->catalog_mgr()->GetRootCatalog()->FindNested(p, &hash, &size);
```

Comment:
Let's discuss this once more.  I think `FindNested` alone is not sufficient, as the correct nested catalog might not be attached yet.  Being a public method now, it would also need to apply the appropriate lock (`ReadLock` or `WriteLock`). I think we need to wrap the `FindNested` call after all, in a public `LookupNested` method.